repose-aggregator/services/service-client/impl/src/main/java/org/openrepose/core/services/serviceclient/akka/impl/AkkaServiceClientImpl.java

```diff
@@ -72,27 +74,42 @@ public AkkaServiceClientImpl(HttpClientService httpClientService) {
                 .expireAfterWrite(FUTURE_CACHE_TTL, FUTURE_CACHE_UNIT)
                 .build();
 
-
-        tokenActorRef = actorSystem.actorOf(new Props(new UntypedActorFactory() {
-            public UntypedActor create() {
-
-                return new AuthTokenFutureActor(serviceClient);
-            }
-        }).withRouter(new RoundRobinRouter(numberOfActors)), "authRequestRouter");
+        createTokenActor();
     }
 
-
     @PreDestroy
     public void destroy() {
-        //Tie this actor system to our bean lifecycle
+        // tie this actor system to our bean lifecycle
         actorSystem.shutdown();
     }
 
     @Override
+    public void setConnectionPoolId(String connectionPoolId) {
+        if (StringUtils.equals(this.connectionPoolId, connectionPoolId)) {
+            return;
+        }
+
+        ActorRef oldTokenActorRef = tokenActorRef;
```

Comment:
You don't want to recreate the actor, you want to change the state in the actor, and then change the size of the pool of actors used for routing.